great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from typing import Any, Dict, Iterable, List, Optional, Union
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.batch import BatchRequest, RuntimeBatchRequest
+from great_expectations.rule_based_profiler.parameter_builder.parameter_builder import (
+    ParameterBuilder,
+)
+from great_expectations.rule_based_profiler.types import (
+    Domain,
+    ParameterContainer,
+    build_parameter_container,
+)
+from great_expectations.validator.validator import Validator
+
+logger = logging.getLogger(__name__)
+
+
+class SimpleDateFormatStringParameterBuilder(ParameterBuilder):
+    """
+    Detects the domain date format from a set of candidate date format strings by computing the
+    column_values.match_strftime_format.unexpected_count metric for each candidate format and returning the format that
+    has the lowest unexpected_count ratio.
+    """
+
+    CANDIDATE_STRINGS = {
+        "%Y-%m-%d",
+        "%m-%d-%Y",
+        "%y-%m-%d",
+        "%Y-%m-%dT%z",
+        "%Y-%m-%d %H:%M:%S",
+    }
+
+    def __init__(
+        self,
+        name: str,
+        metric_domain_kwargs: Optional[Union[str, dict]] = None,
```

Comment:
@NathanFarmer I believe that we also need `     metric_value_kwargs: Optional[Union[str, dict]] = None,`.  Thanks.

Reply:
✅